Sport1Player/Sport1PlayerAdapter.m

```diff
@@ -318,24 +328,95 @@ -(void)shouldPresentPin {
         }
     }];
 }
-#pragma mark - Livestream Token
--(NSObject <ZPPlayable>*)amendIfLivestream:(NSObject <ZPPlayable>*)current {
-    if (current.isLive) {
-        NSString *streamToken = [[[ZAAppConnector sharedInstance] storageDelegate] sessionStorageValueFor:kTokenName
-                                                                                                namespace:kNameSpace];
-        
-        if (streamToken == nil) {return current;}
-        
-        NSString *url = current.contentVideoURLPath;
-        NSString *amendedURL = [[NSString alloc] initWithFormat:@"%@?access_token=%@", url, streamToken];
-        Sport1StreamPlayable *amended = [[Sport1StreamPlayable alloc] initWithOriginal:current
-                                                                         andAmendedURL:amendedURL];
-        return amended;
-    } else {return current;}
-}
+
 #pragma mark - Handlers
+
 -(void)applicationWillEnterForegroundNotificationHandler {
     [self shouldPresentPin];
 }
 
+#pragma mark - Livestream Token
+
+- (void)amendIfLivestreamModified:(NSObject <ZPPlayable>*)current
+                         callback:(void (^)(NSObject <ZPPlayable>* amended))completion {
+
+    if (!current.isLive) {
+        completion(current);
+        return;
+    }
+
+    [Sport1PlayerAdapter getStreamTokenFromAPI:^(BOOL success, NSString *streamToken) {
+        if (!success || !streamToken) {
+            completion(current);
+            return;
+        }
+
+        NSString *amendedURL = [NSString stringWithFormat:@"%@?access_token=%@", current.contentVideoURLPath, streamToken];
+
+        Sport1StreamPlayable *amended =
+        [Sport1StreamPlayable.new initWithOriginal:current
+                                     andAmendedURL:amendedURL];
+
+        completion(amended);
+    }];
+}
+
++ (void)getStreamTokenFromAPI:(void (^)(BOOL success, NSString *streamToken))completion {
+
+    NSString *userToken = [[ZPLoginManager.sharedInstance createWithUserData] getUserToken];
+
+    if (!userToken) {
+        completion(NO, nil);
+        return;
+    }
+
+    NSURL *url = [NSURL URLWithString:@"https://services.inplayer.com/items/71614/access"];
```

Comment:
The identifier 71614 should be taken from the model. It is coming in the extensions dictionary under the key auth_id, and it is 2 nubmers separed by a _ (71614_66734), you should take just the first part.

Reply:
Hi @PRuedaApplicaster , i'm trying to get it from (NSObject <ZPPlayable>* ) but it's empty.

Reply:
There is only one value in the extensionsDictionary:
{
    free = 0;
}

Reply:
You can use the extensionsDictionary of the ZPPlayable object. Look the screenshot:
![Screenshot 2019-10-24 at 11 55 10](https://user-images.githubusercontent.com/29577423/67479077-2aaacb00-f655-11e9-9a1f-6ed2edaf3dc6.png)

Reply:
I am getting more values in the extensionsDictionary. What video are you using, or how are you testing it?